lib/DDG/Meta/Data.pm

```diff
@@ -122,6 +125,29 @@ sub get_ia {
     return $m;
 }
 
+sub _satisfy {
+	my ($ia, $by, $lookup) = @_;
+	return $by->($ia, $lookup) if ref $by eq 'CODE';
+	my $pby = $ia->{$by};
+	ref $lookup eq 'CODE' ? $lookup->($pby) : $pby eq $lookup;
+}
+
+# filter_ias({ repo => 'goodies', dev_milestone => 'live'... })
+# Lookups combine as an AND operation.
+sub filter_ias {
+	my ($lookups) = @_;
+	my @results = @ia_container;
+	while (my ($by, $lookup) = each %$lookups) {
```

Comment:
No, you can stop processing checks on the first failure.  Maybe not a big enough difference to worry too much about.